tests/loan_protocol.rs

```diff
@@ -472,22 +380,38 @@ async fn lend_and_dynamic_liquidate() {
 
     let borrower = borrower.interpret(SECP256K1, loan_response).unwrap();
     let loan_transaction = borrower
-        .sign(|transaction| async move { Ok(borrower_wallet.sign_all_inputs(transaction)) })
+        .sign({
+            let wallet = wallet.clone();
+            |tx| async move {
+                let wallet = wallet.lock().unwrap();
+                Ok(wallet.sign_inputs(tx))
+            }
+        })
         .await
         .unwrap();
 
     let loan_transaction = lender
         .finalise_loan(loan_transaction, {
-            let client = client.clone();
-            |transaction| async move { client.sign_raw_transaction(&transaction).await }
+            let wallet = wallet.clone();
+            |tx| async move {
+                let wallet = wallet.lock().unwrap();
+                Ok(wallet.sign_inputs(tx))
+            }
```

Comment:
If you wouldn't `move` here, you should be able to get away with not cloning either which might allow you to get rid of `Arc<Mutex>>`.

Reply:
Great point. I was really hoping we could get rid of most of these.

It can be done except for those API calls which use the wallet twice: for coin selection and signing. These closures are probably going away soon anyway, so the situation should improve.

Reply:
We only need to use `Arc<Mutex<Wallet>>` once!